src/main/java/org/localhost/wmsemployee/exceptions/handler/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,111 @@
+package org.localhost.wmsemployee.exceptions.handler;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.validation.ConstraintViolationException;
+import lombok.extern.slf4j.Slf4j;
+import org.localhost.wmsemployee.exceptions.EmployeeNotFoundException;
+import org.localhost.wmsemployee.exceptions.NotAValidSupervisorException;
+import org.localhost.wmsemployee.exceptions.NotValidTokenException;
+import org.localhost.wmsemployee.exceptions.PasswordNotValidException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.method.annotation.MethodArgumentTypeMismatchException;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@RestControllerAdvice
+@Slf4j
+public class GlobalExceptionHandler {
+
+    @ExceptionHandler(ConstraintViolationException.class)
+    public ResponseEntity<ErrorResponse> handleValidationExceptions(ConstraintViolationException ex, HttpServletRequest request) {
+
+        log.debug("validation error: {}", ex.getMessage());
+
+        Map<String, String> errors = new HashMap<>();
+
+        ex.getConstraintViolations().forEach(violation -> {
+            String propertyPath = violation.getPropertyPath().toString();
+            String message = violation.getMessage();
+            errors.put(propertyPath, message);
+        });
+
+        ErrorResponse errorResponse = new ErrorResponse(HttpStatus.BAD_REQUEST, "Validation failed", request.getRequestURI(), errors);
+
+        return new ResponseEntity<>(errorResponse, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(org.springframework.web.method.annotation.HandlerMethodValidationException.class)
+    public ResponseEntity<ErrorResponse> handleHandlerMethodValidationException(org.springframework.web.method.annotation.HandlerMethodValidationException ex, HttpServletRequest request) {
+
+        log.debug("Method parameter validation error: {}", ex.getMessage());
+
+        Map<String, String> errors = new HashMap<>();
+
+        ex.getParameterValidationResults().forEach(result -> result.getResolvableErrors().forEach(error -> {
+            String paramName = result.getMethodParameter().getParameterName();
+            String message = error.getDefaultMessage();
+            errors.put(paramName != null ? paramName : "param", message);
+        }));
+
+        ErrorResponse errorResponse = new ErrorResponse(HttpStatus.BAD_REQUEST, "Parameter validation failed", request.getRequestURI(), errors);
+
+        return new ResponseEntity<>(errorResponse, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(MethodArgumentTypeMismatchException.class)
+    public ResponseEntity<ErrorResponse> handleMethodArgumentTypeMismatch(MethodArgumentTypeMismatchException ex, HttpServletRequest request) {
```

Comment:
jak masz np body niepoprawne to spring sam ci zwróci błąd odpowiedni czyli tam 400